src/utils.js

```diff
@@ -64,13 +112,17 @@ export function equals(a, b) {
  */
 export async function byteLengthFromUrl(url, requestInit, customFetch) {
   const fetch = customFetch ?? globalThis.fetch
-  return await fetch(url, { ...requestInit, method: 'HEAD' })
-    .then(res => {
-      if (!res.ok) throw new Error(`fetch head failed ${res.status}`)
-      const length = res.headers.get('Content-Length')
-      if (!length) throw new Error('missing content length')
-      return parseInt(length)
-    })
+  const res = await fetch(url, { ...requestInit, method: 'HEAD' })
+
+  // If HEAD request is forbidden (common with signed S3 URLs), try GET with range
+  if (res.status === 403) {
+    return byteLengthFromUrlUsingGet(url, requestInit, fetch)
+  }
+
+  if (!res.ok) throw new Error(`fetch head failed ${res.status}`)
+  const length = res.headers.get('Content-Length')
```

Comment:
Note that the server has no obligation to provide this header in a HEAD request (the [specification](https://httpwg.org/specs/rfc9110.html#HEAD) says that it MIGHT include it).

So: instead of throwing, we might want to call `byteLengthFromUrlUsingGet` here

Reply:
Done